rpc/jsonrpc_client.go

```diff
@@ -157,20 +157,16 @@ func (cli *JSONRPCClient) GenerateTransactionManual(
 	// Construct transaction
 	now := time.Now().UnixMilli()
 	rules := parser.Rules(now)
-	base := &chain.Base{
-		Timestamp: utils.UnixRMilli(now, rules.GetValidityWindow()),
-		ChainID:   rules.ChainID(),
-		MaxFee:    maxFee,
-	}
+	timestamp := utils.UnixRMilli(now, rules.GetValidityWindow())
+	tx := chain.NewTx(timestamp, rules.ChainID(), maxFee, actions)
```

Comment:
Constructing the transaction here and then modifying it in the loop below may incorrectly mutate the transaction.

Previously, each modifier ran on `base` before it was passed into the transaction, so that it was constructed correctly with the final value.

Now that the modifiers are run after the transaction has been constructed, it will mutate the field without updating the private fields that depend on it.

See `Digest()` as an example: https://github.com/ava-labs/hypersdk/blob/main/chain/transaction.go#L49

It looks like the required order of operations is to construct the transaction and then call Digest/Sign.

Reply:
Nice catch, thank you !
I'm wondering if the fix would be to give a reference to every fields that used to belong to base instead of the full transaction to avoid footguns.

Reply:
Fixed in https://github.com/ava-labs/hypersdk/pull/1054/commits/92b1b776a1de50c38d82a1150ca2bd3ee0744dc8
We got rid of the Base but the 3 fields are always hanging together loosely, not sure if it's desirable